Add simple usage example to README.

jcsackett opened the pull request.

jcsackett committed:
Add docs to make check.

jcsackett committed:
Ignore share dir.

jcsackett committed:
Quick usage docs added to README.

jcsackett commented:
QA:

Verify that the example provided actually works for you.

jujugui commented:
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/theblues-lib/6/
Test PASSed.

fabricematrat commented on Makefile:
```diff
@@ -11,7 +11,7 @@ help:
-	@echo "check - run tests against 2.7 and 3.4 and check lint."
+	@echo "check - clean env, run tests against 2.7 and 3.4, check lint,"
 	@echo "test-all - run tests against 2.7 and 3.4, check lint, and test docs."
 	@echo "coverage - check code coverage quickly with the default Python"
```
I think it's test-coverage or you should rename the target in the Makefile.

jcsackett replied:
That is indeed a mismatch--thanks for spotting it. I've renamed the target to "coverage", to match targets like "check", "lint", etc.

fabricematrat commented on README.rst:
```diff
@@ -16,3 +16,37 @@ which must be installed from the repos::
+charmstore (e.g. a charm or bundle)::
+
+    >>> from theblues.charmstore import CharmStore
+    >>> cs = CharmStore('http://api.jujucharms.com/v4')
```
I would put https here.

jcsackett replied:
Sure thing. Done.

fabricematrat commented on README.rst:
```diff
@@ -16,3 +16,37 @@ which must be installed from the repos::
+    ...
+    <snipped for length>
+    ...
+    juju-log "So, environment is setup. We'll wait for some hooks to fire off before we get all crazy"
```
:)

jcsackett replied:
The best part is that's the actual text from the hook. :P I kind of love that they wrote that in their install hook.

fabricematrat commented:
:+1:  with minor comments.
For future ref, I think we need to think not only about methods docs but also the data format so people knows where to find things.

bac commented on Makefile:
```diff
@@ -11,7 +11,7 @@ help:
 	@echo "lint - check style with flake8"
 	@echo "test - run tests quickly with the default Python"
-	@echo "check - run tests against 2.7 and 3.4 and check lint."
+	@echo "check - clean env, run tests against 2.7 and 3.4, check lint,"
```
why the final comma? none of the others have trailing punctuation.

jcsackett replied:
That's weird--the actual file reads "clean env, run tests against 2.7 and 3.4, check lint and test docs." It does have a trailing period, which I've removed. Not sure why the diff is clipping that last part.

bac commented on README.rst:
```diff
@@ -16,3 +16,37 @@ which must be installed from the repos::
 	$apt-get install libmacaroons0 python-macaroons libsodium13
```
These packages come from a PPA.  I think it should be mentioned.

jcsackett replied:
I did not know that. Added, and thanks for catching that.

bac commented:
:+1: and they all work!  QA-OK

jcsackett commented:
@fabricematrat thanks for the review! I've addressed your comments, I think.

> For future ref, I think we need to think not only about methods docs but also the data format so people knows where to find things.

I concur. I'm hoping we have public charmstore docs at some point covering that we can point to from theblues docs, rather than reproducing all of that information, but either way we need it.

This also raises the point that our library right now is little more than a wrap around a standard GET on the charmstore--at some point we may want to consider a higher level module that provides a nicer, more intuitive object.

jcsackett commented:
@bac Thanks for the review and the QA.

jujugui commented:
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/theblues-lib/7/
Test PASSed.

jcsackett commented:
Thanks again @fabricematrat and @bac. :shipit:

jujugui commented:
Status: merge request accepted. Url: http://ci.jujugui.org:8080/job/theblues-lib-merge

jujugui commented:
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/theblues-lib/8/
Test PASSed.

jujugui commented:
Build failed: Attempt to land pull request failed
        build url: http://ci.jujugui.org:8080/job/theblues-lib-merge/8

jujugui commented:
Status: merge request accepted. Url: http://ci.jujugui.org:8080/job/theblues-lib-merge
